src/main/java/com/coveros/selenified/utilities/Property.java

```diff
@@ -0,0 +1,268 @@
+     *                storing app url information
+     * @return String: the URL of the application under test
+     */
+    // TODO - should we align nomenclature? Either AppUrl (from CMD) or TestSite (from java code)
+    public static String getAppURL(String clazz, ITestContext context) throws InvalidHTTPException {
+        String appURL = null;
+        appURL = checkAppURL(appURL, (String) context.getAttribute(clazz + APP_URL), "The provided app via test case setup '");
+        Properties prop = new Properties();
+        try (InputStream input = new FileInputStream(SELENIFIED)) {
+            prop.load(input);
+        } catch (IOException e) {
+            log.info(e);
+        }
+        appURL = checkAppURL(appURL, prop.getProperty(APP_URL), "The provided app via Properties file '");
+        appURL = checkAppURL(appURL, System.getProperty(APP_URL), "The provided app via System Properties '");
+        if (appURL != null && !"http://".equals(appURL)) {
+            return appURL;
+        }
+        throw new InvalidHTTPException("There was not a valid app provided to test. Please properly set the 'appURL'");
+    }
+
+    /**
+     * A helper method to getAppURL, which checks the provided URL, and if it is valid, overrides the initially
+     * provided one.
+     *
+     * @param originalAppURL - the original and currently set app url
+     * @param newAppURL      - the new app url to check
+     * @param s              - the location being checked (for reporting)
+     * @return String: the most valid URL, new if it is valid, original if not
+     */
+    private static String checkAppURL(String originalAppURL, String newAppURL, String s) {
+        if (newAppURL != null) {
+            if (!newAppURL.toLowerCase().startsWith("http") && !newAppURL.toLowerCase().startsWith("file")) {
+                newAppURL = "http://" + newAppURL;
+            }
+            try {
+                new URL(newAppURL);
+                return newAppURL;
+            } catch (MalformedURLException e) {
+                log.error(s + newAppURL + "' is not a valud URL.");
+            }
+        }
+        return originalAppURL;
+    }
+
+    /**
+     * Retrieves the browser property if it is set. This can be a single browser name, or browser details. If it is
+     * not set, HTMLUnit will be returned as the default browser to use
+     *
+     * @return String: the set browser
+     */
+    public static String getBrowser() {
+        String browser = getProgramProperty(BROWSER);
+        if (browser == null || "".equals(browser)) {
+            browser = Browser.BrowserName.HTMLUNIT.toString();
+        }
+        return browser;
+    }
+
+    /**
+     * Determines if the headless parameter was set, to have the browser run in headless mode. This only
+     * can be used for Chrome and Firefox.
+     *
+     * @return boolean: is headless set or not
+     */
+    public static boolean headless() {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Rename method "headless" to prevent any misunderstanding/clash with field "HEADLESS" defined on line 57. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1845)